plugins/user-authenticators/ldap/src/org/apache/cloudstack/ldap/LdapManagerImpl.java

```diff
@@ -57,17 +57,22 @@
     private LdapContextFactory _ldapContextFactory;
 
     @Inject
-    private LdapUserManager _ldapUserManager;
+    private LdapConfiguration _ldapConfiguration;
+
+    @Inject LdapUserManagerFactory _ldapUserManagerFactory;
+
 
     public LdapManagerImpl() {
         super();
     }
 
-    public LdapManagerImpl(final LdapConfigurationDao ldapConfigurationDao, final LdapContextFactory ldapContextFactory, final LdapUserManager ldapUserManager) {
+    public LdapManagerImpl(final LdapConfigurationDao ldapConfigurationDao, final LdapContextFactory ldapContextFactory, final LdapUserManagerFactory ldapUserManagerFactory,
+                           final LdapConfiguration ldapConfiguration) {
```

Comment:
It was initially done that way. We could inject as well. I did that in the new class.

-----Original Message-----
From: "Daan Hoogland" <notifications@github.com>
Sent: ‎22-‎07-‎2015 13:18
To: "apache/cloudstack" <cloudstack@noreply.github.com>
Cc: "Rajani Karuturi" <rajanikaruturi@gmail.com>
Subject: Re: [cloudstack] CLOUDSTACK-8596 ability to query nested groups forMicrosoft AD (#609)

In plugins/user-authenticators/ldap/src/org/apache/cloudstack/ldap/LdapManagerImpl.java:
>  
>      public LdapManagerImpl() {
>          super();
>      }
>  
> -    public LdapManagerImpl(final LdapConfigurationDao ldapConfigurationDao, final LdapContextFactory ldapContextFactory, final LdapUserManager ldapUserManager) {
> +    public LdapManagerImpl(final LdapConfigurationDao ldapConfigurationDao, final LdapContextFactory ldapContextFactory, final LdapUserManagerFactory ldapUserManagerFactory,
> +                           final LdapConfiguration ldapConfiguration) {
And why not use injection for that as well? Is this a groovy thing?
—
Reply to this email directly or view it on GitHub.